### ISSUE TITLE
`query xml` should return full nodes including attributes

### ISSUE THREAD

**Issue by develop7** [nushell/nushell#12562]
### Describe the bug

`query xml` outputs, um, a plain table losing resulting nodes' attributes. This behavior doesn't match the `from xml` result layout as well, while it should IMO.

### How to reproduce

1.  Having the `query` plugin installed, run the following:
```nu
❯  '<zju><foo bar="baz">yarr</foo></zju>' | query xml "/zju/foo"
╭───┬──────────╮
│ # │ /zju/foo │
├───┼──────────┤
│ 0 │ yarr     │
╰───┴──────────╯
```

### Expected behavior

At least node attributes should be preserved for certain, `from xml` does that for one

```nu
❯  '<zju><foo bar="baz">yarr</foo></zju>' | from xml | get content.0
╭────────────┬────────────────────────────────────╮
│ tag        │ foo                                │
│            │ ╭─────┬─────╮                      │
│ attributes │ │ bar │ baz │                      │
│            │ ╰─────┴─────╯                      │
│            │ ╭───┬─────┬────────────┬─────────╮ │
│ content    │ │ # │ tag │ attributes │ content │ │
│            │ ├───┼─────┼────────────┼─────────┤ │
│            │ │ 0 │     │            │ yarr    │ │
│            │ ╰───┴─────┴────────────┴─────────╯ │
╰────────────┴────────────────────────────────────╯
```

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.92.1                                                            |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.77.0 (aedd173a2 2024-03-17) (built from a source tarball) |
| cargo_version      | cargo 1.77.0                                                      |
| build_time         | 2024-04-04 16:48:53 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | dataframe, default, sqlite, trash, which                          |
| installed_plugins  | formats, gstat, inc, query                                        |

**Comment by fdncred** [comment]
It just uses xpath, so I'm not sure that's possible. Please feel free to submit a PR to make it better. XML is very difficult to deal with, I know.

**Comment by develop7** [comment]
> It just uses xpath, so I'm not sure that's possible.

True, but it also stringifies resulting nodes, dropping everything except node content, including attributes.

**Comment by fdncred** [comment]
I'd support a flag to allow this other information to be returned but I don't think I'd just make the switch. From my experience with xpath, which may be different than others, I always want the results in a list of data. That's probably because my most frequent xpath search is `//something`.

**Comment by develop7** [comment]
> I'd support a flag to allow this other information to be returned but I don't think I'd just make the switch.

Yup, it's backward-compatible approach. Makes sense.

**Comment by j-xella** [comment]
How about an `--xml` switch, which would produce results as strings containing XML representations of xpath matches?  In this case, since `/zju/foo` matches a node, the result would be

```nu
[ '<foo bar="baz">yarr</foo>' ]
```

That would at least give users something to work with, for example pipe the results through `from xml` ...
